### PR TITLE
Fixes #8: Added method and example to show importance of various data points in train_features

### LINKED ISSUE #8: Importance score for dataset training samples
Implement a way to assess the importance of an inidividual training datapoint to the performance of the model. This could be done by training the same model with a particular point included and then excluded from the training set, and computing the difference in performance scores on the test set.

### PR DESCRIPTION
**Pull request by KaairaGupta**
Following changes have been made in this pr:
- Under dev/KaairaGupta/importance_score, added function `imp_score_singular_datapoints` in file importance_score.py with following specifications:
  - Inputs: 
            - model: model object which must have function model.fit(train_features, train_labels)
            - train_features: numpy array of feature values for training
            - train_labels: numpy array of correct labels for train_features
            - test_features: numpy array of feature values for test data
            - test_labels: numpy array of correct labels for test data
            - i_start: start index to remove features from train_labels (one-by-one)
            - i_end: end index (not inclusive) to remove features from train_labels (one-by-one)
  - Output:
            - Plot of difference in performance scores (accuracy and f-score) on the test set by removing data points (one-by-one) from training data.
- Added example to use this function in example_importance_score.ipynb

### CODE REVIEW

File: dev/KairaGupta/importance_score/importance_score.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+            i_start: start index to remove features from train_labels (one-by-one)
+            i_end: end index (not inclusive) to remove features from train_labels (one-by-one)
+        Output:
+            Plot of difference in performance scores (accuracy and f-score) on the test set by removing data points (one-by-one) from training data.
+    """
+    
+    fs_list = []
+    acc_list = []
+    
+    temp_model = model
```
Comment (mlopatka) [inline dev/KairaGupta/importance_score/importance_score.py]: nit: full model (or rather model trained with full data availability)
resulting in: `fs_full_data` and `acc_full_data` 
could have been trained inside the `for` loop resulting in less code clutter.

Not by any means a critical change, just a note on code cleanliness.

File: dev/KairaGupta/importance_score/importance_score.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    pred_full_data = temp_model.predict(test_features)
+
+    fs_full_data = metrics.f1_score(test_labels, pred_full_data)
+    acc_full_data = metrics.accuracy_score(test_labels, pred_full_data)
+    
+    for i in range(i_start,i_end):
+        temp_train = np.concatenate((train_features[:i], train_features[i+1:]))
+        temp_labels = np.concatenate((train_labels[:i], train_labels[i+1:]))
+        temp_model = model
+        temp_model.fit(temp_train, temp_labels)
```
Comment (mlopatka) [inline dev/KairaGupta/importance_score/importance_score.py]: nit: this line could be directly called passing `np.concatenate((train_features[:i], train_features[i+1:]` and `np.concatenate((train_labels[:i], train_labels[i+1:]))` and still be very readable.
I understand that assigning to a new variable on each iteration looks cleaner, but there is a fine line between readability and unnecessary operations.

Again, this PR is solid and I dot really have any complaints, these are just styling preferences.

File: dev/KairaGupta/importance_score/importance_score.py (hunk 1 above)
Comment (KaairaGupta) [reply dev/KairaGupta/importance_score/importance_score.py]: Sure! I will make these changes ASAP! Thanks

File: dev/KairaGupta/importance_score/importance_score.py (hunk 2 above)
Comment (KaairaGupta) [reply dev/KairaGupta/importance_score/importance_score.py]: I'll keep this in mind in my future PRs as well. Thanks!

**Comment by KaairaGupta** [comment]
@mlopatka how do I merge this??

File: dev/KairaGupta/importance_score/importance_score.py (hunk 2 above)
Comment (KaairaGupta) [reply dev/KairaGupta/importance_score/importance_score.py]: @mlopatka If I wish to keep it as it for now (I would like to work on other issues) and make changes in future, how do I merge this (as you have told in above comment)?

Comment (mlopatka) [reply dev/KairaGupta/importance_score/importance_score.py]: oh! sorry for the delayed reply. This PR was already approved.
I'll merge it in now. 
In the future, if a PR is approved you can also feel free to merge it into master.
